### PR TITLE
[JBTM-3645] Made it clear that ObjStoreBrowser should be considered singleton

### PR DESCRIPTION
**Pull request by jmfinelli**
Addresses [JBTM-3645](https://issues.redhat.com/browse/JBTM-3645)

!CORE !TOMCAT !AS_TESTS !RTS !JACOCO !XTS !QA_JTA !QA_JTS_OPENJDKORB !PERF !LRA !DB_TESTS !mysql !db2 !postgres !oracle

### CODE REVIEW

**Comment by jmfinelli** [comment]
CORE failed because of the change of API in `ObjStoreBrowser` (which this JIRA/PR is all about). Should I deprecate the constructors of `ObjStoreBrowser` and remove them when we are sure that the community is aware of this modification?

**Review by mmusgrov — COMMENTED** [review]
I added a comment to the issue indicating that making an API change for very little benefit is questionable - the whole package is already marked as `Deprecated` so there is no requirement to support ad hoc usage of the instrumentation. I also pointed out the knock on effect of breaking existing management applications that use the instrumentation.

**Comment by mmusgrov** [comment]
@jmfinelli I've added the hold label until you've decided how you wish to proceed. As discussed elsewhere this is a change that breaks certain JMX/management expectations and is a change that we do not want.

**Comment by jmfinelli** [comment]
That's fine to me. Maybe, let's just add a comment to the javadoc of `ObjStoreBrowser` to make it clear that, by design, there must not be two instances of `ObjStoreBrowser` around (basically, developers should consider `ObjStoreBrowser` as a singleton class).

I would also add that we should consider when it is a good time to move `ObjStoreBrowser` into an internal package (maybe next major release?) and add it to our schedule.

I will copy this comment to the related JIRA to keep track of this conversation. Thanks Michael :-)

File: ArjunaCore/arjuna/classes/com/arjuna/ats/arjuna/tools/osb/mbean/ObjStoreBrowser.java [hunk 1]
```diff
@@ -303,14 +303,26 @@ private void init(String logDir) {
...
         this(null);
     }
 
-    public ObjStoreBrowser(String logDir) {
+     /**
+     * <p>This is the constructor to configure ObjStoreBrowser with a path to load the Object Store.</p>
+     *
+     * <p>This class has been designed with the assumption that only one instance of ObjStoreBrowser should be used.
+     * In other words, this class should be considered as a singleton class. In fact, if two ObjStoreBrowser instances
+     * are created, they will interere with each other.</p>
```
Comment (mmusgrov) [inline ArjunaCore/arjuna/classes/com/arjuna/ats/arjuna/tools/osb/mbean/ObjStoreBrowser.java:323]: /will interere/may interfere/

**Review by mmusgrov — CHANGES_REQUESTED** [review]
Just a couple of typos but otherwise okay.

File: ArjunaCore/arjuna/classes/com/arjuna/ats/arjuna/tools/osb/mbean/ObjStoreBrowser.java [hunk 2]
```diff
@@ -303,14 +303,26 @@ private void init(String logDir) {
...
     }
 
+
     /**
-     * Constructor for default configuration of the object store browser.
+     * <p>This is the constructor for the default configuration of ObjStoreBrowser.</p>
+     *
+     * <p>This class has been designed with the assumption that only one instance of ObjStoreBrowser should be used.
+     * In other words, this class should be considered as a singleton class. In fact, if two ObjStoreBrowser instances
+     * are created, they will interere with each other.</p>
```
[the remark this answers is not in the document]
Comment (jmfinelli) [reply ArjunaCore/arjuna/classes/com/arjuna/ats/arjuna/tools/osb/mbean/ObjStoreBrowser.java]: Thanks Michael. Corrected +1

**Review by mmusgrov — APPROVED** [review]

**Pull request merged**